src/BitStringAddresses/fermifs.jl

```diff
@@ -1,57 +1,106 @@
 """
     FermiFS{N,M,S} <: SingleComponentFockAddress
 
-Address type that represents a Fock state of `N` fermions of the same spin in `M` modes
-by wrapping a bitstring of type `S <: BitString`.
+Address type that represents a Fock state of `N` fermions of the same spin in `M` modes by
+wrapping a `[`BitString`](@ref), or a [`SortedParticleList`](@ref). Which is wrapped is
+chosen automatically based on the properties of the address.
 
 # Constructors
 
-* `FermiFS{N,M}(bs::BitString)`: Unsafe constructor. Does not check whether the number of
-  particles in `bs` is equal to `N`.
+* `FermiFS{[N,M]}(onr)`: Create `FermiFS{N,M}` from [`onr`](@ref) representation. This is
+  efficient if `N` and `M` are provided, and `onr` is a statically-sized collection, such as
+  a `Tuple{M}` or `SVector{M}`.
 
-* `FermiFS(::BitString)`: Automatically determine `N` and `M`. This constructor is not type
-  stable!
+* `FermiFS{[N,M]}([M, ]pairs...)`: Provide the number of modes `M` and pairs of the form
+  `mode => 1`. If `M` is provided as a type parameter, it should not be provided as the
+  first argument.  Useful for creating sparse addresses. `pairs` can be multiple arguments
+  or an iterator of pairs.
 
-* `FermiFS{[N,M,S]}(onr)`: Create `FermiFS{N,M}` from [`onr`](@ref) representation. This is
-  efficient as long as at least `N` is provided.
+* `FermiFS{N,M,S}(bs::S)`: Unsafe constructor. Does not check whether the number of
+  particles in `bs` is equal to `N`, or whether each mode only contains one particle.
+
+* [`@fs_str`](@ref): addresses are sometimes printed in a compact manner. This
+  representation can also be used as a constructor. See the last example below.
 
 See also: [`SingleComponentFockAddress`](@ref), [`BoseFS`](@ref), [`BitString`](@ref).
+
+# Examples
+
+```jldoctest
+julia> FermiFS{3,5}((0, 1, 1, 1, 0))
+FermiFS{3,5}((0, 1, 1, 1, 0))
+
+julia> FermiFS([abs(i - 3) ≤ 1 for i in 1:5])
+FermiFS{3,5}((0, 1, 1, 1, 0))
+
+julia> FermiFS(5, 2 => 1, 3 => 1, 4 => 1)
+FermiFS{3,5}((0, 1, 1, 1, 0))
```

Comment:
It would be nice if the following
`FermiFS(5, 2 => 1, 3 => 1, 4 => -1)`
automatically built a two-component address. Would that be difficult?

Reply:
Neat idea, I'll implement it for `FermiFS2C`.